frontend/packages/kubevirt-plugin/src/components/vm-templates/vm-template-create-yaml.tsx

```diff
@@ -22,13 +23,38 @@ const CreateVMTemplateYAMLConnected = connectToPlural(
       }
       return <ErrorPage404 />;
     }
+    const [selectedTemplate, setTemplate] = React.useState(null);
+    const [templateLabel, setTemplateLabel] = React.useState(null);
 
     const template = VMTemplateYAMLTemplates.getIn(['vm-template']);
     const obj = safeLoad(template);
     obj.kind = kindObj.kind;
     obj.metadata = obj.metadata || {};
     obj.metadata.namespace = match.params.ns || 'default';
 
+    if (templateLabel) {
+      obj.objects[0].metadata.labels['vm.kubevirt.io/template'] = templateLabel;
+    }
+
+    if (!selectedTemplate) {
+      k8sGet(TemplateModel, '', 'openshift')
+        .then((templates) => {
+          setTemplate(
+            templates.items.filter(
+              (item) =>
+                item.metadata.labels['flavor.template.kubevirt.io/medium'] &&
+                item.metadata.labels['workload.template.kubevirt.io/server'] &&
+                item.metadata.name.includes('fedora'),
+            )[0],
+          );
+        })
+        .catch((err) => console.log('err:', err));
+    }
+
+    if (selectedTemplate && !templateLabel) {
+      setTemplateLabel(selectedTemplate.objects[0].metadata.labels['vm.kubevirt.io/template']);
+    }
+
```

Comment:
The rerendering of the new label does not work. Either use skeleton before the templates load or use the props of `EditYAML_` correctly. Maybe `sampleObj` or empty `obj`? 

Note: In case you want to use sampleObj:  sampleObj  should be saved in the state of vm-template-create-yaml, so that the reference doesn't change and  this doesn't trigger multipleLoading of the yaml and performance issues,

Reply:
or just use the loading straight ahead and render edit yaml async component later